pkg/lib/config/configsource/kubernetes.go

```diff
@@ -155,7 +156,13 @@ func (k *Kubernetes) invalidateHostMap(ingress *networkingv1beta1.Ingress) {
 }
 
 func (k *Kubernetes) updateHostMap(appID string, ingress *networkingv1beta1.Ingress) {
-	for _, host := range extractIngressHosts(ingress) {
+	hosts := extractIngressHosts(ingress)
+
+	k.ingressMap.Store(ingress.UID, &ingressSnapshot{
```

Comment:
We should delete the previous hosts, for example, when ingress hosts are updated from `["test1.localhost"]` to `["test2.localhost"]`, `test1.localhost` should be deleted.

Reply:
fixed.